### DISCUSSION TITLE
Congratulations  and some ideea, reports

### DISCUSSION THREAD

**Discussion by claww** [BimmerGestalt/AAIdrive#205] [Ideas]
Hello,

I would like to congratulation for the initiative, that's a very good thing what you have done here., i always wonder why they have not implementing more apps as bmw comes with this nice feature, however I like more to use the idrive instead of android auto, i would prefer to have apps coming from the phone like in the way you did.

I would like to report somethings as well and some features if can be implemented...

My car have idrive id5,  when i use youtube music, after some plays, it show disconnected in the song name, but song still play and im unable to change the songs.
Youtube would be nice if its possible to make it play the videos as well, otherwise is still like youtube music.

Keep up the nice work !

**Comment by hufman** [comment]
Thank you for the kind words! The tight integration is my favorite aspect of this protocol.

Can you give more information about what leads up to YouTube Music losing connection? I just tried playing a playlist for an hour and it stayed connected the whole time.
Sadly, both the Android MediaController and the car's protocol are both oriented towards static pictures, so even if there was a way to get the YouTube stream, the car's API wouldn't satisfactorily show it.

**Reply by claww** [reply]
Hey,

I notice this happen when i change the songs, but if i go back in application and select again the youtube music it connects back and show the song name instead of disconnected, but in this whole time the song is not interrupted
Ill can try make a video and send it to you if im not so clear

**Reply by bogdan-calapod** [reply]
Hey @hufman :D I also noticed this sometimes, but it doesn't happen very often so I didn't think much about it. It seems like YT Music is somehow taken out of memory, but the music still plays. I've noticed that other media control apps, for example my Galaxy Watch, can't control it when it "borks out". 

Re-entering the app or keeping it in the foreground fixes it, so I think it's an Android memory issue. I did notice it far less after I've disabled battery/memory optimizations in the Android settings.

**Reply by claww** [reply]
At me is happen very often, also the search is not working, when i refer at reentering in app i refer from idrive, not from phone. 

I just saw, there is a new verision of app, I was using the release 1.2.2, ill lets you know if the problem persists with the latest version

Thanks

**Reply by claww** [reply]
After i installed latest release, problem is gone. Thanks!
